include/nighthawk/adaptive_load/adaptive_load_controller.h

```diff
@@ -10,26 +10,32 @@
 namespace Nighthawk {
 
 /**
- * Performs an adaptive load session consisting of the Adjusting Stage and the
- * Testing Stage. Adjusting Stage: Runs a series of short benchmarks, checks metrics according to
- * MetricSpecs, and adjusts load up or down based on the result; returns an error if convergence is
- * not detected before the deadline in the spec. Load adjustments and convergence detection are
- * computed by a StepController plugin. Metric values are obtained through MetricsPlugins. Testing
- * Stage: When the optimal load is found, runs one long benchmark to validate it.
- *
- * @param nighthawk_service_stub A Nighthawk Service gRPC stub.
- * @param spec A proto that defines all aspects of the adaptive load session, including metrics,
- * threshold, duration of adjusting stage benchmarks, and underlying Nighthawk traffic parameters.
- * @param time_source An abstraction of the system clock. Normally, just construct an
- * Envoy::Event::RealTimeSystem and pass it. NO_CHECK_FORMAT(real_time). If calling from an
- * Envoy-based process, there may be an existing TimeSource or TimeSystem to use. If calling
- * from a test, pass a fake TimeSource.
- *
- * @return StatusOr<AdaptiveLoadSessionOutput> A proto logging the result of all traffic attempted
- * and all corresponding metric values and scores, or an overall error status if the session failed.
+ * Contains the main loop of the adaptive load controller. Consults a StepController for load
+ * decisions, interacts with Nighthawk Service and MetricsPlugins.
  */
-absl::StatusOr<nighthawk::adaptive_load::AdaptiveLoadSessionOutput> PerformAdaptiveLoadSession(
-    nighthawk::client::NighthawkService::StubInterface* nighthawk_service_stub,
-    const nighthawk::adaptive_load::AdaptiveLoadSessionSpec& spec, Envoy::TimeSource& time_source);
+class AdaptiveLoadController {
+public:
+  virtual ~AdaptiveLoadController() = default;
+  /**
+   * Performs an adaptive load session consisting of the Adjusting Stage and the
+   * Testing Stage. Adjusting Stage: Runs a series of short benchmarks, checks metrics according to
+   * MetricSpecs, and adjusts load up or down based on the result; returns an error if convergence
+   * is not detected before the deadline in the spec. Load adjustments and convergence detection are
+   * computed by a StepController plugin. Metric values are obtained through MetricsPlugins. Testing
+   * Stage: When the optimal load is found, runs one long benchmark to validate it.
+   *
+   * @param nighthawk_service_stub A Nighthawk Service gRPC stub.
```

Comment:
nit: Maybe stub is more appropriate as a member in the class than an argument to the method. I don't think we expect it to change on a multiple call scenario.

Reply:
This class is intended to be stateless helpers and it's only a class to make mocking easier. We will have a chance to design a convenient client when we merge `RemoteProcessImpl` and `NighthawkServiceClientImpl`.